### ISSUE TITLE
PageRemove attribute

### ISSUE THREAD

**Issue by miladahmadi33** [dotnet/aspnetcore#39864]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

```[PageRemote]``` attribute in ```POST``` method doesn't send ```__requestverificationtoken``` to server and I get error 400.

### Expected Behavior

```PageRemote``` attribute doesn't work and doesn't send ```__requestverificationtoken``` filed on post method.
This is my ViewModel
`
public class AddCategory
{
    [PageRemote(PageName = "Seller", PageHandler = "CheckForTitle",
        HttpMethod = "POST",
        AdditionalFields = "__RequestVerificationToken",
        ErrorMessage = "This title is duplicate")]
    public string Title { get; set; } = null!;
}
`
And this is my handler
`public class CategoryModel : PageModel
    {
        [BindProperty]
        public AddCategory Category { get; set; }

        public void OnGet()
        {
        }

        public IActionResult OnPostCheckTitle(AddCategory category)
        {
            return new JsonResult(category.Title == "a");
        }
    }`
```GET``` method is ok and everything is fine, but in ```POST``` method ```__requestverificationtoken``` doesn't send to the server and I get error 400.

### .NET Version

6.0.101

**Comment by ShreyasJejurkar** [comment]
@pranavkm I can reproduce this issue with current .NET 6 as well. 
I have found the root cause as well ( I guess so ). 

The issue in `jquery.validate.unobtrusive.js` file that we import for validation purposes. 

I tried to debug the JS file which is minified to find a cause. 
![image](https://user-images.githubusercontent.com/17148381/153759223-0b15264e-d0b7-4027-bd9f-89f4a4acf0f6.png)

The issue is above on the line where my current debugger is. If you see it's trying to find control and its corresponding value so that that can be sent as payload to the server. For other properties, it works fine if you see below. My property was Email in Student Model with Student object name as StudentData. As the JS is minified, not sure why we are adding `\\` in between property names. See below in the image. But still, it can be able to find correct control and grab its value. 
![image](https://user-images.githubusercontent.com/17148381/153759283-ecb99f5a-77f8-43b2-a341-2179cf29bf1e.png)

The problem arises for __RequestForgeryToken, it cannot able to find the correct DOM input control. I doubt is it something to when there `\\` and `__` simultaneously. 
![image](https://user-images.githubusercontent.com/17148381/153759393-a6fc8d1f-dac9-4980-86a3-89a63d03f561.png)

**Comment by ShreyasJejurkar** [comment]
Yes, I guess there is some issue when we have field with `__` at the start. See below, I deliberately added __TestValue to the model and the page. But that did not work. The value was not picked by jquery when doing remote requests. 
![image](https://user-images.githubusercontent.com/17148381/153759948-25165259-3bc9-4d71-af59-fa853eb53c4c.png)

The request. 
![image](https://user-images.githubusercontent.com/17148381/153759971-4f9b25ad-f8fd-4cd9-b0b7-8c69e2cf243c.png)

The model 
![image](https://user-images.githubusercontent.com/17148381/153759999-90632bfd-de21-4ec2-8d2d-2fb4cd0b8823.png)

**Comment by ShreyasJejurkar** [comment]
Where can I find the actual source code of `jquery.validate.unobtrusive.js` so that I can debug it much easier? Debugging minified files of JS is not so much fun! :) @pranavkm

**Comment by yankarinRG** [comment]
> Where can I find the actual source code of `jquery.validate.unobtrusive.js` so that I can debug it much easier? Debugging minified files of JS is not so much fun! :) @pranavkm

Here is the source code: https://github.com/aspnet/jquery-validation-unobtrusive/blob/main/src/jquery.validate.unobtrusive.js 🙂
